go/vt/vttablet/tabletserver/vstreamer/copy.go

```diff
@@ -243,14 +244,21 @@ func (uvs *uvstreamer) copyTable(ctx context.Context, tableName string) error {
 				log.V(2).Infof("Not starting fastforward pos is %s, uvs.pos is %s, rows.gtid %s", pos, uvs.pos, rows.Gtid)
 			}
 
+			// Store a copy of the fields and pkfields because the original will be cleared
+			// when GRPC returns our request to the pool
+			uvs.fields = slice.Map(rows.Fields, func(f *querypb.Field) *querypb.Field {
+				return f.CloneVT()
+			})
+			uvs.pkfields = slice.Map(rows.Pkfields, func(f *querypb.Field) *querypb.Field {
+				return f.CloneVT()
+			})
```

Comment:
just nit picking here, we could extract the anonymous function into a variable and reuse it in both `slice.map` calls.

Reply:
That would actually (sadly, devastatingly, disappointingly) prevent inlining, which is why I copied the body twice right now. :sob: